sdk/objectanchors/Azure.MixedReality.ObjectAnchors/src/ObjectAnchorsClient.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.MixedReality.Authentication;
+using System.Threading.Tasks;
+
+namespace Azure.MixedReality.ObjectAnchors
+{
+    /// <summary>
+    /// Represents a client for Azure Object Anchors.
+    /// </summary>
+    public class ObjectAnchorsClient
+    {
+        private readonly Guid _accountId;
+
+        private readonly HttpPipeline _pipeline;
+
+        private readonly BlobUploadEndpointRestClient _getBlobUploadEndpointRestClient;
+
+        private readonly IngestionJobRestClient _ingestionJobRestClient;
+
+        internal readonly ClientDiagnostics _clientDiagnostics;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="token">An access token used to access the specified Azure Object Anchors account.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, AccessToken token)
+            : this(accountId, accountDomain, new StaticAccessTokenCredential(token), new ObjectAnchorsClientOptions()) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
+        /// </summary>
+        /// <param name="accountId">The Azure Object Anchors account ID.</param>
+        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
+        /// <param name="token">An access token used to access the specified Azure Object Anchors account.</param>
+        /// <param name="options">The options.</param>
+        public ObjectAnchorsClient(Guid accountId, string accountDomain, AccessToken token, ObjectAnchorsClientOptions options)
+            : this(accountId, accountDomain, new StaticAccessTokenCredential(token), options) { }
```

Comment:
```suggestion
        /// <summary>
        /// Initializes a new instance of the <see cref="ObjectAnchorsClient" /> class.
        /// </summary>
        /// <param name="accountId">The Azure Object Anchors account ID.</param>
        /// <param name="accountDomain">The Azure Object Anchors account domain.</param>
        /// <param name="token">An access token used to access the specified Azure Object Anchors account.</param>
        /// <param name="options">The options.</param>
        public ObjectAnchorsClient(Guid accountId, string accountDomain, AccessToken token, ObjectAnchorsClientOptions options)
            : this(accountId, accountDomain, new StaticAccessTokenCredential(token), options = null) { }
```

In the other libraries, we've removed this constructor in favor of making the `options` parameter optional in the other constructors.

Made this comment in [ApiView](https://apiview.dev/Assemblies/Review/1fdf4c5c1109482ca6316d34319903e9/0c2de770563e44abab4c2f28abbc4fa6?doc=False) as well, but it appears to have been ignored.

Reply:
Sorry, I must have missed it or thought it was a question to the arch board who originally proposed that we have a no-options constructor, which I chose to extend to each constructor so all means of authentication benefitted from this approachability. If we want to choose a single constructor to apply this approach to, we can do that and go the options optional in the rest. 
(See remarks at 33:55 [here](https://msit.microsoftstream.com/video/f603a4ff-0400-b564-fe5b-f1eb6b2e4cd0?referrer=https:%2F%2Fstatics.teams.cdn.office.net%2F))

Reply:
The changes I've suggested here, which were suggested to me in the Authentication library, offer the exact same constructor options, but you only declare 3 instead of 6. The API remains the same, but it's far less code by making the `options` parameter optional.

Reply:
Reduced to a single constructor with optional options